src/ex2/main.js

```diff
@@ -0,0 +1,89 @@
+class Main {
+  constructor() {
+    this.addButten = null;
+    this.sortButton = null;
+    this.input - null;
+    this.itemManager = new ItemManager();
+    this.pokemonClient = new PokemonClient();
+    this.inputHandler = new Input();
+    this.htmlEditor = new HtmlEditor(this.itemManager, this.pokemonClient);
+  }
+
+  async handleNewInput() {
+    let answer = this.inputHandler.handleAndValidateInput();
+    if (!answer) {
+      alert("An empty task is for lazy pepole ;)");
+      return;
+    }
+    let { newPokemonsIdArr, newNormalTasks } = answer;
+    this.inputHandler.handleAndValidateInput();
+    if (newPokemonsIdArr === undefined || newNormalTasks === undefined) {
+      return;
+    }
+    let normalTasksToBeAdd =
+      this.itemManager.handleNewNormalTasks(newNormalTasks);
+
+    let pokemonsNameToBeFetch =
+      this.pokemonClient.checkForPokemonInNoramlTasks(normalTasksToBeAdd);
```

Comment:
Main should not know pokemonClient as well.
main should initialize application and activate DOM readiness, also populating it (using itemManager)

if tomorrow you'll change the functionality to be a digimon todos? you'll need to refactor both `main` and  `htmlEditor`, instead of just refactoring `itemManager`